src/rules/if_change_then_change.rs

```diff
@@ -24,55 +25,83 @@ lazy_static::lazy_static! {
 
 }
 
-pub fn ictc(hunks: &Vec<Hunk>) -> anyhow::Result<Vec<diagnostic::Diagnostic>> {
+pub fn find_ictc_blocks(path: &PathBuf) -> anyhow::Result<Vec<IctcBlock>> {
+    let mut blocks: Vec<IctcBlock> = Vec::new();
+
+    let mut ifttt_begin: i64 = -1;
+
+    let in_file = File::open(path).with_context(|| format!("failed to open: {:#?}", path))?;
+    let in_buf = BufReader::new(in_file);
+
+    for (i, line) in lines_view(in_buf)
+        .context(format!("failed to read lines of text from: {:#?}", path))?
+        .iter()
+        .enumerate()
+        .map(|(i, line)| (i + 1, line))
+    {
+        if RE_BEGIN.find(line).is_some() {
+            ifttt_begin = i as i64;
+        } else if let Some(end) = RE_END.captures(line) {
+            if ifttt_begin != -1 {
+                let block = IctcBlock {
+                    path: path.clone(),
+                    begin: ifttt_begin,
+                    end: i as i64,
+                    thenchange: PathBuf::from(end.get(2).unwrap().as_str()),
```

Comment:
Nit. Use context and `?` operator as above. e.g. `.unwrap().` => `.wich_context(... "expected at least 3 captures")?.`